### PR TITLE
fix(DB): Evening/Nighttime Game Events Update

### LINKED ISSUE #7587: [Westfall] Several issues with Leprithus
### What client do you play on?

enUS

### Faction

- [X] Alliance
- [X] Horde

### Content Phase:

- [ ] Generic
- [X] 1-19
- [ ] 20-29
- [ ] 30-39
- [ ] 40-49
- [ ] 50-59

### Current Behaviour
Original report: https://github.com/chromiecraft/chromiecraft/issues/1559

Leprithus (https://wowgaming.altervista.org/aowow/?npc=572)  has roaming in north spawn point, and is currently on 20h spawn timer, and is missing patrol route in his southern spawn point.

Testers note:
Currently, all his spawn points are wandering with a distance of 5.

### Expected Blizzlike Behaviour

Leprithus should be stationary in his northern spawn.
Leprithus should have a longer patrol route in southern spawn.

As for spawn time, there are two comments (1 from vanilla and one from classic) that mention he spawns 8.00 PM server time every day, but there also are comments that meniton him being around during the day - likely he wasn't killed before and stuck around.

By carsono311 on 2019/09/09 (Classic)
1. Leprithus is in game and spawns at approximately 8:00 pm server time. This is identical to his original spawn time in Vanilla.

2. When I encountered him today, he spawned at the northern point near Jangolode Mine, however I had camped that spot previously at that time and he never spawned... As such, I suspect that he is spawning at only one spot (either the northern spawn or the southern) each day at that time, which is not how he behaved in Vanilla. Originally, he would spawn in BOTH locations at the same time. While a little weird, it was in the game. It seems that Blizzard have removed this oddity with Classic.

3. He still spawns with 2 Ghouls near him. At the northern spawn, he is stationary (inside the little graveyard) unless aggroed. At the southern spawn, he will roam with his Ghoul buddies until aggroed/killed

And a comment from vanilla TBC and then edited during cata:

By Hazu on 2007/11/03 (Patch 2.2.3)
This is rumour control. Here are the facts:

Leprithus and his two Rotten Ghoul companions spawn at the same time every day: approximately 8:00 pm server time. Do not confuse his spawn time with that of the ghouls in the Moonbrook graveyard - those ghouls spawn in place of the normal buzzards at 9:00 pm daily, when game "day" turns to game "night". Since Leprithus spawns an hour earlier, it's safe to assume that these two incidents are not connected or dependent upon one another.

With the coming of Deathwing, this particular rare spawn lost the quality that once made him so unusual - the fact that he was a "double spawn". Pre-Cataclysm, another copy of Leprithus and his pals would also spawn at the same time in a small defile in the hills south of the Dust Plains, at 56,80 (also in a small three-grave cemetery).

Unlike their stationary northern twins, the southern Leprithus and his ghouls would path northeast across the Dust Plain, towards the covered bridge that leads from Westfall to Duskwood. If they made it to the bridge unopposed by the old People's Militia patrol, they would stop for a few moments, then commence wandering about in the area between the road and the mage tower, and through some of the nearby knoll camps.

Leprithus now only spawns at the northern location - 42,29 in the small graveyard. His two unique green drops, the Ghoulfang sword and the Ghoul Fingers gloves, are still on his loot table, but I've yet to see either item drop since the launch of Cataclysm (or appear in the auction house).

Testers note:
Coparing the map from [wowgaming](https://wowgaming.altervista.org/aowow/?npc=572) and [Wowhead](https://tbc.wowhead.com/npc=572/leprithus#map), it's easy to see that something is off:
image
image

Further explanation of routes can be found here: https://wowpedia.fandom.com/wiki/Leprithus?oldid=2089314
This also states that there can be multiple occurrences simultaneously, tho, I doubt the validity of that claim.

### Source

https://tbc.wowhead.com/npc=572/leprithus

### Steps to reproduce the problem

1. teleport to Leprithus (entry 572)
2. Observe him moving in northern spawn
3. Observe him roaming in southern spawn points(but not a proper patrol route)

### Extra Notes

_No response_

### AC rev. hash/commit

https://github.com/chromiecraft/azerothcore-wotlk/commit/d20f315bd5370761f226460964ae506a97f372c0

### Operating system

Ubuntu 20.04

### Modules

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chromie-xp](https://github.com/azerothcore/mod-chromie-xp)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [lua-carbon-copy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-custom-corldboss](https://github.com/55Honey/Acore_CustomWorldboss)
- [lua-level-up-reward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-recruit-a-friend](https://github.com/55Honey/Acore_RecruitAFriend)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-temp-announcements](https://github.com/55Honey/Acore_TempAnnouncements)
- [lua-zonecheck](https://github.com/55Honey/acore_Zonecheck)

### Customizations

None

### Server

ChromieCraft

### PR DESCRIPTION
**Pull request by LannyE**
Ever wondered why Miss Danna has her school children out looking at buildings in the middle of the night?  Doesn't Adam and Billy's fathers wonder where they are?  In vanilla wow they despawned at night.  

This PR will add evening and night events into Azerothcore.  Here are the changes:
Pyrewood Village Event Start Time fix -> Was too early
Moonrage Sentry Faction fix ->  Not hostile to alliance at night
Ultham Ironhorn -> Move to corral at night (Pyrewood Village Event)
Ultham's Rams -> Move to corral at night (Pyrewood Village Event)
Frost and Black Ram not available in WOTLK changed to correct Ram
Rotten Ghouls spawn during (Pyrewood Village) night event
Fleshrippers replace Rotten Ghouls when night event not active
Defias Smugglers replace Rotten Ghouls when night not active
Eastvale Children movement and locations adjusted
Evening Game event added -> Stars at 6:00 server time
Eastvale Children despawn during Evening event
Following Stormwind Civilians also despawn during evening event
Adam and Billy
Justin, Brandon, and Roman
Miss Danna and school children
Donna and William
Janey, Suzanne, and Lisan
Orphans that are running around at Orphanage
Karlee Chaddis, Paige Chaddis, and Gil
New Leprithus Event -> Starts at 8:00 server time
2 Leprithus spawn in different loactions in WOTLK at 8:00
Southern Graveyard has waypoints added
2 Rotten Ghoul minions spawn with each Leprithus

Some info for this update was pulled from TC database.

## Changes Proposed:
-  See above
-  

## Issues Addressed:

- Closes #7587

## SOURCE:

Here are some sources used:
https://wowpedia.fandom.com/wiki/Nocturnal
https://tbc.wowhead.com/npc=572/leprithus#comments

## Tests Performed:

- Watched each NPC spawn or despawn during events.  
- Followed Leprithus around and tested his formations.

## How to Test the Changes:

1. At 6:00 Server time go to Stormwind and see if the following civilians despawn:
Adam and Billy
Justin, Brandon, and Roman
Miss Danna and school children
Donna and William
Janey, Suzanne, and Lisan
Orphans that are running around at Orphanage
Karlee Chaddis, Paige Chaddis, and Gil
2. At 6:00 PM go to Eastvale Logging camp and see if they are despawned
3. At 8:00 PM check for Leprithus GUID 134020 and 134021.  They should both spawn.  The southern has waypoints
4. At 9:00 PM check for Ultham Ironhorn GUID 76265 has moved to in front of the corral
5. At 9:00 PM check for Ultham Ironhorn's ram have moved into the corral
6. At 9:00 PM check for Rotten Ghouls replacing fleshrippers and smugglers in graveyard in Moonbrook and next to Jangolode mine.
7. At 9:00 PM check Pyrewood village for them to convert to Moonrage Worgen. 
8. At 9:00 PM Check Moonrage Sentry's are hostile to alliance players
9.  At 6:00 AM all Stormwind civilians, Fleshrippers, smugglers, Pyrewood Village should return to normal.
10. At 6:00 AM any Leprithus that are still alive will despawn.

## Known Issues and TODO List:

- [The times for these events are based on websites in the source section and other obscure blogs.  If someone has more accurate info please let me know.]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by LannyE** [comment]
I can't fix the clang issue with creature GUID 87023 (GEOFF) until the Stormwind update PR gets merged.  The missing GUID is in that PR.

File: data/sql/updates/pending_db_world/rev_1648344440179668500.sql [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+(-88,79816),
+(-88,79817),
+(-88,87092), -- Miss Danna and students
+(-88,87088),
+(-88,87090),
+(-88,87089),
+(-88,87082),
+(-88,87091),
+(-88,84028),
+(-88,87023),
```
Comment (acidmanifesto) [inline data/sql/updates/pending_db_world/rev_1648344440179668500.sql:108]: image

**Review by acidmanifesto — CHANGES_REQUESTED** [review]

**Comment by acidmanifesto** [comment]
> I can't fix the clang issue with creature GUID 87023 (GEOFF) until the Stormwind update PR #11121 gets merged. The missing GUID is in that PR.

just merged it. will rerun the tests

File: data/sql/updates/pending_db_world/rev_1648344440179668500.sql (hunk 1 above)
Comment (acidmanifesto) [reply data/sql/updates/pending_db_world/rev_1648344440179668500.sql:108]: i will approve once the re-ran tests show green. thanks for the heads up that this needed the other pr.

**Review by acidmanifesto — APPROVED** [review]

**Comment by acidmanifesto** [comment]
@LannyE locally on my end it shows fine. however i think we need you to do a fetch upstream so the tests can rerun with the appropiately revision. https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/syncing-a-fork

**Pull request merged**